provisioning_client/tests/common_prov_e2e/prov_hsm/x509_msr.c

```diff
@@ -86,52 +89,37 @@ static RIOT_X509_TBS_DATA X509_ROOT_TBS_DATA = {
     { 0x1A, 0x2B, 0x3C, 0x4D, 0x5E }, HSM_CA_CERT_NAME, "AZURE_TEST", "US",
     "170101000000Z", "370101000000Z", HSM_CA_CERT_NAME, "AZURE_TEST", "US" };
 
-// The "root" signing key. This is intended for development purposes only.
-// This key is used to sign the DeviceID certificate, the certificiate for
-// this "root" key represents the "trusted" CA for the developer-mode
-// server(s). Again, this is for development purposes only and (obviously)
-// provides no meaningful security whatsoever.
-static unsigned char eccRootPubBytes[sizeof(ecc_publickey)] = {
-    0xeb, 0x9c, 0xfc, 0xc8, 0x49, 0x94, 0xd3, 0x50, 0xa7, 0x1f, 0x9d, 0xc5,
-    0x09, 0x3d, 0xd2, 0xfe, 0xb9, 0x48, 0x97, 0xf4, 0x95, 0xa5, 0x5d, 0xec,
-    0xc9, 0x0f, 0x52, 0xa1, 0x26, 0x5a, 0xab, 0x69, 0x00, 0x00, 0x00, 0x00,
-    0x7d, 0xce, 0xb1, 0x62, 0x39, 0xf8, 0x3c, 0xd5, 0x9a, 0xad, 0x9e, 0x05,
-    0xb1, 0x4f, 0x70, 0xa2, 0xfa, 0xd4, 0xfb, 0x04, 0xe5, 0x37, 0xd2, 0x63,
-    0x9a, 0x46, 0x9e, 0xfd, 0xb0, 0x5b, 0x1e, 0xdf, 0x00, 0x00, 0x00, 0x00,
-    0x00, 0x00, 0x00, 0x00 };
-
-static unsigned char eccRootPrivBytes[sizeof(ecc_privatekey)] = {
-    0xe3, 0xe7, 0xc7, 0x13, 0x57, 0x3f, 0xd9, 0xc8, 0xb8, 0xe1, 0xea, 0xf4,
-    0x53, 0xf1, 0x56, 0x15, 0x02, 0xf0, 0x71, 0xc0, 0x53, 0x49, 0xc8, 0xda,
-    0xe6, 0x26, 0xa9, 0x0b, 0x17, 0x88, 0xe5, 0x70, 0x00, 0x00, 0x00, 0x00 };
-
-static int generate_root_ca_info(X509_CERT_INFO* x509_info, RIOT_ECC_SIGNATURE* tbs_sig)
+static int generate_root_ca_info(X509_CERT_INFO* x509_info)
 {
     int result;
     uint8_t der_buffer[DER_MAX_TBS] = { 0 };
     DERBuilderContext der_ctx = { 0 };
     DERBuilderContext der_pri_ctx = { 0 };
     RIOT_STATUS status;
-
-    memcpy(&x509_info->ca_root_pub, eccRootPubBytes, sizeof(ecc_publickey));
-    memcpy(&x509_info->ca_root_priv, eccRootPrivBytes, sizeof(ecc_privatekey));
+    RIOT_ECC_SIGNATURE tbs_sig = { 0 };
 
     // Generating "root"-signed DeviceID certificate
     DERInitContext(&der_ctx, der_buffer, DER_MAX_TBS);
     DERInitContext(&der_pri_ctx, der_buffer, DER_MAX_TBS);
 
-    if (X509GetDeviceCertTBS(&der_ctx, &X509_ROOT_TBS_DATA, &x509_info->ca_root_pub) != 0)
+    if ((status = RiotCrypt_DeriveEccKey(&x509_info->ca_root_pub, &x509_info->ca_root_priv,
```

Comment:
Same as above: where does the key-pair come from?

Reply:
This shouldn't happen, there is no root CA key on a device (derived or otherwise).  This wouldn't exist in this context outside of a reference or sample code.

Reply:
x509_msr.c is part of the e2e/sample, however this question is the same as the one for the production code

Reply:
Same comments here from above

Reply:
same response to comments above, this one is our e2e/sample so we could force the use of p256 for the curve if we want to compile time define the key bytes

Reply:
Forced NIST 256, and baked-in a Dev CA Root - Fixed